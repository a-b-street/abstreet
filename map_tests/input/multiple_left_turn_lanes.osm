<?xml version='1.0' encoding='UTF-8'?>
<!-- Resembles https://www.openstreetmap.org/node/53187904 with multiple left turn lanes. -->
<osm>
        <bounds minlon="0.0" maxlon="0.01" minlat="0.0" maxlat="0.01"/>
        <node id="1" lon="0.005" lat="0.005"/>
        <node id="2" lon="0.005" lat="-1.0"/>
        <node id="3" lon="0.005" lat="1.0"/>
        <node id="4" lon="-0.1" lat="0.005"/>
        <node id="5" lon="1.0" lat="0.005"/>
        <way id="100">
            <nd ref="1"/>
            <nd ref="2"/>
            <tag k="name" v="south"/>
            <tag k="highway" v="primary"/>
            <tag k="sidewalk" v="both"/>

            <tag k="lanes" v="3"/>
            <tag k="oneway" v="yes"/>
        </way>
        <way id="101">
            <nd ref="3"/>
            <nd ref="1"/>
            <tag k="name" v="north"/>
            <tag k="highway" v="primary"/>
            <tag k="sidewalk" v="both"/>

            <tag k="lanes" v="3"/>
            <tag k="oneway" v="yes"/>
            <tag k="turn:lanes" v="left|left;through|none"/>
        </way>
        <way id="102">
            <nd ref="4"/>
            <nd ref="1"/>
            <tag k="name" v="west"/>
            <tag k="highway" v="residential"/>
            <tag k="sidewalk" v="both"/>

            <tag k="lanes" v="4"/>
            <tag k="oneway" v="yes"/>
	    <tag k="turn:lanes" v="|||right"/>
        </way>
        <way id="103">
            <nd ref="1"/>
            <nd ref="5"/>
            <tag k="name" v="east"/>
            <tag k="highway" v="residential"/>
            <tag k="sidewalk" v="both"/>

            <tag k="lanes" v="3"/>
            <tag k="oneway" v="yes"/>
        </way>
</osm>
