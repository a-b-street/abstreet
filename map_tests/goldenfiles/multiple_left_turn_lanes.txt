TurnID(Lane #0, Lane #4, Intersection #0) is a Crosswalk
TurnID(Lane #0, Lane #20, Intersection #0) is a SharedSidewalkCorner
TurnID(Lane #4, Lane #0, Intersection #0) is a Crosswalk
TurnID(Lane #4, Lane #15, Intersection #0) is a SharedSidewalkCorner
TurnID(Lane #5, Lane #9, Intersection #0) is a Crosswalk
TurnID(Lane #5, Lane #16, Intersection #0) is a SharedSidewalkCorner
TurnID(Lane #6, Lane #17, Intersection #0) is a Left
TurnID(Lane #6, Lane #18, Intersection #0) is a Left
TurnID(Lane #6, Lane #19, Intersection #0) is a Left
TurnID(Lane #7, Lane #1, Intersection #0) is a Straight
TurnID(Lane #7, Lane #2, Intersection #0) is a Straight
TurnID(Lane #7, Lane #3, Intersection #0) is a Straight
TurnID(Lane #7, Lane #17, Intersection #0) is a Left
TurnID(Lane #7, Lane #18, Intersection #0) is a Left
TurnID(Lane #7, Lane #19, Intersection #0) is a Left
TurnID(Lane #8, Lane #1, Intersection #0) is a Straight
TurnID(Lane #8, Lane #2, Intersection #0) is a Straight
TurnID(Lane #8, Lane #3, Intersection #0) is a Straight
TurnID(Lane #8, Lane #17, Intersection #0) is a Left
TurnID(Lane #8, Lane #18, Intersection #0) is a Left
TurnID(Lane #8, Lane #19, Intersection #0) is a Left
TurnID(Lane #9, Lane #5, Intersection #0) is a Crosswalk
TurnID(Lane #9, Lane #10, Intersection #0) is a SharedSidewalkCorner
TurnID(Lane #10, Lane #9, Intersection #0) is a SharedSidewalkCorner
TurnID(Lane #10, Lane #15, Intersection #0) is a Crosswalk
TurnID(Lane #11, Lane #1, Intersection #0) is a Right
TurnID(Lane #11, Lane #2, Intersection #0) is a Right
TurnID(Lane #11, Lane #3, Intersection #0) is a Right
TurnID(Lane #11, Lane #17, Intersection #0) is a Straight
TurnID(Lane #11, Lane #18, Intersection #0) is a Straight
TurnID(Lane #11, Lane #19, Intersection #0) is a Straight
TurnID(Lane #12, Lane #1, Intersection #0) is a Right
TurnID(Lane #12, Lane #2, Intersection #0) is a Right
TurnID(Lane #12, Lane #3, Intersection #0) is a Right
TurnID(Lane #12, Lane #17, Intersection #0) is a Straight
TurnID(Lane #12, Lane #18, Intersection #0) is a Straight
TurnID(Lane #12, Lane #19, Intersection #0) is a Straight
TurnID(Lane #13, Lane #1, Intersection #0) is a Right
TurnID(Lane #13, Lane #2, Intersection #0) is a Right
TurnID(Lane #13, Lane #3, Intersection #0) is a Right
TurnID(Lane #13, Lane #17, Intersection #0) is a Straight
TurnID(Lane #13, Lane #18, Intersection #0) is a Straight
TurnID(Lane #13, Lane #19, Intersection #0) is a Straight
TurnID(Lane #14, Lane #1, Intersection #0) is a Right
TurnID(Lane #14, Lane #2, Intersection #0) is a Right
TurnID(Lane #14, Lane #3, Intersection #0) is a Right
TurnID(Lane #15, Lane #4, Intersection #0) is a SharedSidewalkCorner
TurnID(Lane #15, Lane #10, Intersection #0) is a Crosswalk
TurnID(Lane #16, Lane #5, Intersection #0) is a SharedSidewalkCorner
TurnID(Lane #16, Lane #20, Intersection #0) is a Crosswalk
TurnID(Lane #20, Lane #0, Intersection #0) is a SharedSidewalkCorner
TurnID(Lane #20, Lane #16, Intersection #0) is a Crosswalk
